base/refpointer.jl

```diff
@@ -126,6 +126,17 @@ cconvert(::Type{Ref{P}}, a::Array{<:Ptr}) where {P<:Ptr} = a
 cconvert(::Type{Ptr{P}}, a::Array) where {P<:Union{Ptr,Cwstring,Cstring}} = Ref{P}(a)
 cconvert(::Type{Ref{P}}, a::Array) where {P<:Union{Ptr,Cwstring,Cstring}} = Ref{P}(a)
 
+
+cconvert(::Type{Ref{T}}, t::NTuple{N,T}) where {N,T} = Ref{NTuple{N,T}}(t)
+cconvert(::Type{Ptr{T}}, t::NTuple{N,T}) where {N,T} = Ref{NTuple{N,T}}(t)
```

Comment:
```suggestion
```

Reply:
I’m not sure we want this conversion, if it seems to make sense semantically to convert a Tuple to a Ptr

Reply:
Fair enough, can remove that.

Reply:
I guess it makes as much sense as converting an Array to a Ptr?

Reply:
Not really since the `Tuple` doesn't own the memory.

I would be fine to allow converting `Ref` of `NTuple` to `Ptr` of the `eltype` if it is not already allowed.

Reply:
Yeah, it’s not fundamentally invalid to define this, but I think Ref is just generally closer to the right concept so there’s no reason to also define this. IIRC, cconvert to Ptr is generically defined as the identity function (intentionally letting it fall through to the unsafe_convert rules) so the latter case Yichao mentioned may already work. Would be good to see if it does and add tests for it.